JavaScript/Indentation Rules.tmPreferences

```diff
@@ -2,16 +2,23 @@
 <plist version="1.0">
 <dict>
 	<key>scope</key>
-	<string>source.js, source.ts, source.jsx, source.tsx</string>
+	<string>
+		source.js - source.js meta.mapping - source.js meta.sequence,
+		source.jsx - source.jsx meta.mapping - source.jsx meta.sequence,
+		source.ts - source.ts meta.mapping - source.ts meta.sequence,
+		source.tsx - source.tsx meta.mapping - source.tsx meta.sequence
```

Comment:
I don't think we can do this. It will cause the patterns to not be active inside any mapping and there are some very common constructs with function definitions in objects, where it is normal to write statements inside a mapping, e.g.
```js
export default {
    setup() {
        // normal code be here, e.g. a switch statement
    }
}
```

Ultimately, we might need cooperation from the syntax definition to do this properly.

Reply:
And who asked, why people prefer Treesitter?